cicerone/spec/reference.py

```diff
@@ -0,0 +1,117 @@
+"""Reference model for OpenAPI $ref objects.
```

Comment:
Move this file under a new `cicerone/references/` directory. Refactor tests and imports to handle the change.